### PR TITLE
feat(gitlab): add filtering and pagination (#681)

### LINKED ISSUE #518: gitlab: add support for filtering and pagination of projects
After #509, the profile page in the web interface will show all the GitLab projects the user has access to, even those contained in groups/subgroups. This means that we will show many more projects in the UI, and that sometimes we might reach the limit of 100 projects that can be retrieved in a single request to GitLab APIs.

We should:
 - add support for pagination, so that the user can see all the projects he has access to (instead of only the first 100)
 - add support for filtering the projects, so that the user can find a given project quickly

We also need to change the interface of reana-ui accordingly.

### PR DESCRIPTION
**Pull request by mdonadoni**
- feat(gitlab): add search query parameter to filter projects (#681)
- feat(gitlab): add support for paginated project list (#681)
    
    BREAKING CHANGE: The REST API endpoint `gitlab_projects` now includes
    pagination details.

Closes #518

### CODE REVIEW

File: reana_server/utils.py [hunk 1]
```diff
@@ -509,23 +509,30 @@ def _get_gitlab_hook_id(project_id, gitlab_token):
     :param project_id: Project id on GitLab.
     :param gitlab_token: GitLab token.
     """
-    reana_hook_id = None
+    # FIXME: handle pagination of results
```
Comment (tiborsimko) [inline reana_server/utils.py:512]: Do we need the FIXME here?

**Review by tiborsimko — APPROVED** [review]
Works nicely :+1: 

Some cosmetic comments:

- Left a comment about a (forgotten?) FIXME in the code.

- The base commit could perhaps say "add" instead of "accept" (?), giving something like:

```
 feat(gitlab): add search query parameter to filter projects (#681)
```

File: reana_server/utils.py (hunk 1 above)
Comment (mdonadoni) [reply reana_server/utils.py:512]: This is fixed by #684 , but I have removed it anyway

**Comment by mdonadoni** [comment]
Updated the commit title, and also the BREAKING CHANGE notice to read as follows:

BREAKING CHANGE: The REST API endpoint `gitlab_projects` now includes pagination details.

**Pull request merged**